ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -67,6 +76,8 @@ public ObjectInspector initialize(ObjectInspector[] arguments)
         argumentOIs = arguments;
         if (outputCategory == ObjectInspector.Category.LIST) {
             return initListOI(arguments);
+        } else if (outputCategory == ObjectInspector.Category.PRIMITIVE && outputPrimitiveCategory == PrimitiveObjectInspector.PrimitiveCategory.STRING) {
+            return PrimitiveObjectInspectorFactory.writableStringObjectInspector;
```

Comment:
I feel its better to push this down to GenericUDFArrayJoin by overriding initialize in GenericUDFArrayJoin. This condition seems specific to Array join and I dont see this condition used by other potential UDFs.

This will avoid the addition of another constructor as well to this generic class.

Reply:
pushed down to GenericUDFArrayJoin